plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-product-variations-controller.php

```diff
@@ -877,4 +904,30 @@ public function get_collection_params() {
 
 		return $params;
 	}
+
+	/**
+	 * Create all variations for a given product.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_Error|WP_REST_Response
+	 */
+	public function create_all( $request ) {
+		$product_id = (int) $request['product_id'];
+
+		if ( 'product' !== get_post_type( $product_id ) ) {
```

Comment:
Instead of calling `get_post_type( $product_id )` here and then `wc_get_product( $product_id );` on line 925 which means 2 database hits, I prefer to call `wc_get_product( $product_id );` at the very begining and then check  
```php
$product = wc_get_product( $product_id );

if ( isset( $product ) ) ) {
   ...
```

which is gonna reduce the amount of database hits to just 1.